app/com/linkedin/drelephant/spark/heuristics/DriverHeuristic.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2016 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.linkedin.drelephant.spark.heuristics
+
+import java.util.ArrayList
+
+import scala.collection.JavaConverters
+import scala.util.Try
+import com.linkedin.drelephant.analysis._
+import com.linkedin.drelephant.configurations.heuristic.HeuristicConfigurationData
+import com.linkedin.drelephant.spark.data.SparkApplicationData
+import com.linkedin.drelephant.util.MemoryFormatUtils
+import com.linkedin.drelephant.spark.fetchers.statusapiv1.ExecutorSummary
+
+/**
+  * A heuristic based the driver's configurations and memory used. It checks whether the configuration values specified are within the threshold range. It also analyses the peak JVM memory used, unified memory and time spent in GC by the job.
+  *
+  */
+class DriverHeuristic(private val heuristicConfigurationData: HeuristicConfigurationData)
```

Comment:
This heuristic contains multiple checks. Is it possible to add some information about which checks are problematical? Another option could be to split into separate heuristics.